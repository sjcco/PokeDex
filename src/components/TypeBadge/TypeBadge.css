
.pkmn-type {
  font-size: 15px;
  font-family: 'Trebuchet MS', sans-serif;
  font-weight: bold;
  width: 6em;
  text-align: center;
  display: inline-block;
  border: 0.1em solid transparent;
  padding: 0.1em 0.2em 0.2em;
  margin: 0.1em 0.15em;
  border-radius: 0.4em;
}

.pkmn-type span {
  text-transform: uppercase;
  font-size: 0.8em;
  color: white;
  text-shadow: 0 0 0.2em black;
}
