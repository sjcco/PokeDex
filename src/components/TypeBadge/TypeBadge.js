import React from 'react';
import PropTypes from 'prop-types';
import { pkmnType } from './TypeBadge.css';

const TypeBadge = ({ type }) => {
  const typeclass = type => {
    switch (type) {
      case 'normal':
        return '#a8a878';
      case 'fire':
        return '#f08030';
      case 'water':
        return '#6890f0';
      case 'grass':
        return '#98d888';
      case 'flying':
        return '#a890f0';
      case 'fighting':
        return '#c03028';
      case 'poison':
        return '#a040a0';
      case 'electric':
        return '#f8d030';
      case 'ground':
        return '#e0c068';
      case 'rock':
        return '#b8a038';
      case 'psychic':
        return '#f85888';
      case 'ice':
        return '#98d8d8';
      case 'bug':
        return '#a8b820';
      case 'ghost':
        return '#705898';
      case 'steel':
        return '#b8b8d0';
      case 'dragon':
        return '#7038f8';
      case 'dark':
        return '#705848';
      case 'fairy':
        return '#f0b6bc';
      default:
        return null;
    }
  };

  const colorStyle = {
    backgroundColor: typeclass(type),
  };

  return (
    <div className={pkmnType} style={colorStyle}>
      <span>{type}</span>
    </div>
  );
};

TypeBadge.propTypes = {
  type: PropTypes.string.isRequired,
};

export default TypeBadge;
