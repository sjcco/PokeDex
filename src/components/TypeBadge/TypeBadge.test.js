/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TypeBadge from './TypeBadge';

test('renders content', () => {
  const component = renderer.create(<TypeBadge
    type="type"
  />).toJSON();
  expect(component).toMatchSnapshot();
});
