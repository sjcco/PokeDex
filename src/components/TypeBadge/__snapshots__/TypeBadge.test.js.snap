// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="pkmnType"
  style={
    Object {
      "backgroundColor": null,
    }
  }
>
  <span>
    type
  </span>
</div>
`;
